src/agent/communicator/include/ihttp_socket.hpp

```diff
@@ -18,8 +21,10 @@ namespace http_client
         /// @brief Connects the socket to the given endpoints
         /// @param endpoints The endpoints to connect to
         /// @param ec The error code, if any occurred
-        virtual void Connect(const boost::asio::ip::tcp::resolver::results_type& endpoints,
-                             boost::system::error_code& ec) = 0;
+        virtual void Connect(boost::asio::io_context& io_context,
```

Comment:
Add missing parameters to function description.

Reply:
Done